protocol/flows/src/v5/txrelay/flow.rs

```diff
@@ -199,6 +190,29 @@ impl RelayTransactionsFlow {
         }
     }
 
+    /// If in the last 10 seconds we exceeded the TPS threshold, we will throttle tx relay
+    fn check_tx_throttling(&self, throttling_state: &mut ThrottlingState) {
```

Comment:
It's possible to make this function pure - it would accept throttling state, time and snapshot. Then it would be easy to write test without instantiating flow itself.

Reply:
Addressed in [8661512](https://github.com/kaspanet/rusty-kaspa/pull/389/commits/8661512f11677b4941cbf845a38c0b837f38081c)

Although I don't have the time right now to write the tests

Reply:
Right, I don't actually think it has to be a pure function. The important part is that it's now testable without instantiating the class. So I've updated my commit message and added the test.